### PR TITLE
ASP-2101 Patch --exclusive SBATCH parameter

### PR DESCRIPTION
**Pull request by fschuch**
#### What
Patch the `--exclusive` SBATCH parameter.

#### Why
An issue was reported and after further investigation, it was identified that `--exclusive` is a special SBATCH parameter that can be used in two different ways:

1. `--exclusive` as a flag, meaning the value `True` should be recovered for slurmd.
2. `--exclusive=<value>`, meaning the value <value> should be recovered for slurmd. According to the Slurm documentation, the value can be `user` or `mcs`.

`Task`: https://jira.scania.com/browse/ASP-2101

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -450,3 +450,59 @@ def test_bidict__can_be_compared_to_a_dictionary_inverse(self, dummy_mapping):
         desired_value = {value: key for key, value in dummy_mapping.items()}
 
         assert desired_value == bidict(dummy_mapping).inverse
+
+
+class TestExclusiveParameter:
```
Comment (dusktreader) [inline tests/test_parser.py:455]: These are excellent tests. Nice work.

**Review by dusktreader — APPROVED** [review]

**Comment by matheushent** [comment]
Task linked: [ARMADA-438 Discovery: generate tokens for Slurm REST API auth](https://app.clickup.com/t/18022949/ARMADA-438)

File: tests/test_parser.py (hunk 1 above)
Comment (matheushent) [reply tests/test_parser.py:455]: agreed. Very nice job!

**Comment by fschuch** [comment]
Some comments about fixing the `--exclusive` flag on the SBATCH to Slurm API parser here on cluster-agent.

From the SBATCH side, it can be:

1. `#SBATCH --exclusive=mcs`
1. `#SBATCH --exclusive=user`
1. `#SBATCH --exclusive`, this is the one I'm fixing, when it is used as a flag, we should recover a default value.

But which value should we use in the last case? According to slurm documentation, it can be any of the following strings:

1. `user`
1. `mcs`
1. `true`
1. `false`

On the integration test, the two on the top worked with no problem, but the same can not be said about the bottom two. I've tried every value I could imagine that could be used to replace boolean values in this context.

I started poking and googling around, but with no success, until I took a look at slurm's source code, the options from there are:

1. `user`
1. `mcs`
1. `exclusive`
1. `oversubscribe`

All confirmed from my integration tests, and by the use os `srun` command, i.e.:

```bash
srun --exclusive -p compute --pty bash
srun --exclusive=exclusive -p compute --pty bash
srun --exclusive=user -p compute --pty bash
srun --exclusive=mcs -p compute --pty bash
srun --exclusive=oversubscribe -p compute --pty bash
```

**Pull request merged**